packages/react/src/analytics/integrations/Castle/__tests__/Castle.test.ts

```diff
@@ -0,0 +1,236 @@
+import {
+  pageEventsData as analyticsPageData,
+  trackEventsData as analyticsTrackData,
+  loadIntegrationData,
+} from 'tests/__fixtures__/analytics';
+import {
+  eventTypes,
+  integrations,
+  LoadIntegrationEventData,
+  pageTypes,
+  StrippedDownAnalytics,
+  utils,
+} from '@farfetch/blackout-analytics';
+import Castle, { CASTLE_MESSAGE_PREFIX } from '../Castle';
+import isString from 'lodash/isString';
+import pickBy from 'lodash/pickBy';
+import type { CastleIntegrationOptions } from '../types';
+import type { WebEventData } from '../../../types/analytics.types';
+
+const mockRequestHeaderValue = '12342342345241342423424';
+const publishableKey = 'pk_mock_111111111111111111111111111';
+const mockOptions = {
+  pk: publishableKey,
+  debugModeOn: true,
+  window: window,
+  avoidCookies: false,
+  cookieDomain: 'foo',
+  timeout: 1000,
+};
+const createInstance = (
+  options: CastleIntegrationOptions = mockOptions,
+  loadData: LoadIntegrationEventData = loadIntegrationData,
+) => new Castle(options, loadData, {} as StrippedDownAnalytics);
+
+jest.mock('@castleio/castle-js', () => {
+  return {
+    ...jest.requireActual('@castleio/castle-js'),
+    createRequestToken: async () => Promise.resolve(mockRequestHeaderValue),
+  };
+});
+
+// mock this function to avoid installing `jest-canvas-mock` - Castle needs it.
+HTMLCanvasElement.prototype.getContext = jest.fn();
+utils.logger.info = jest.fn();
+
+describe('Castle integration', () => {
+  let instance: Castle | null;
+
+  beforeEach(() => {
+    instance = null;
+
+    jest.clearAllMocks();
+  });
+
+  describe('Castle instance', () => {
+    it('Should extend the core integration class', () => {
+      expect(Castle.prototype).toBeInstanceOf(integrations.Integration);
+    });
+
+    it('Should be an integration that loads independently of user consent', () => {
+      expect(Castle.shouldLoad()).toBe(true);
+    });
+
+    it('Should return a Castle instance from createInstance', () => {
+      instance = createInstance();
+
+      expect(instance).toBeInstanceOf(Castle);
+    });
+  });
+
+  describe('Validation', () => {
+    it('Should log an error if no publishable key is passed via `options`', () => {
+      // @ts-ignore - We want to test this specific case
+      expect(() => createInstance({ pk: null })).toThrowError();
+    });
+
+    it('Should throw an error if an invalid `configureHttpClient` is passed', () => {
+      expect(() =>
+        // @ts-ignore - We want to test this specific case
```

Comment:
Same as here https://github.com/Farfetch/blackout/pull/255/files#r845987247
There are other instances in this file as well which this comment is valid.